cmd/esc/cli/env.go

```diff
@@ -65,49 +66,192 @@ func newEnvCmd(esc *escCommand) *cobra.Command {
 }
 
 type environmentRef struct {
-	orgName string
-	envName string
-	version string
+	orgName     string
+	projectName string
+	envName     string
+	version     string
+
+	hasAmbiguousPath bool
 }
 
 func (r *environmentRef) String() string {
-	s := r.envName
+	s := fmt.Sprintf("%s/%s", r.projectName, r.envName)
+
 	if r.orgName != "" {
-		s = fmt.Sprintf("%s/%s", r.orgName, r.envName)
+		s = fmt.Sprintf("%s/%s", r.orgName, s)
 	}
 	if r.version != "" {
 		s = fmt.Sprintf("%s@%s", s, r.version)
 	}
 	return s
 }
 
-func (cmd *envCommand) getRelativeEnvRef(refStr string, rel *environmentRef) environmentRef {
-	orgName, envNameAndVersion, hasOrgName := strings.Cut(refStr, "/")
-	if !hasOrgName {
-		orgName, envNameAndVersion = cmd.esc.account.DefaultOrg, orgName
+func (cmd *envCommand) parseRef(refStr string) environmentRef {
+	var orgName, projectName, envNameAndVersion string
+
+	hasAmbiguousPath := false
+	orgName = cmd.esc.account.DefaultOrg
+	projectName = client.DefaultProject
+
+	parts := strings.Split(refStr, "/")
+
+	switch l := len(parts); {
+	case l == 1:
+		// <environment-name>
+		envNameAndVersion = parts[0]
+	case l == 2:
+		// <project-name>/<env-name> or <org-name>/<env-name>
+		// We assume the former, and this will be disambiguated later.
+		projectName = parts[0]
+		envNameAndVersion = parts[1]
+
+		hasAmbiguousPath = true
+	case l >= 3:
+		// <org-name>/<project-name>/<environment-name>
+		orgName = parts[0]
+		projectName = parts[1]
+		envNameAndVersion = strings.Join(parts[2:], "")
 	}
 
 	envName, version, hasSep := strings.Cut(envNameAndVersion, "@")
 	if !hasSep {
 		envName, version, _ = strings.Cut(envNameAndVersion, ":")
 	}
 
-	if rel != nil && envName == "" && version != "" {
-		orgName, envName = rel.orgName, rel.envName
+	return environmentRef{
+		orgName:          orgName,
+		projectName:      projectName,
+		envName:          envName,
+		version:          version,
+		hasAmbiguousPath: hasAmbiguousPath,
+	}
+}
+
+// If a non-nil environmentRef is provided, default to its values if an environment name and version are not specified
+func (cmd *envCommand) getEnvRef(refString string, rel *environmentRef) (ref environmentRef, err error) {
+	envRef := cmd.parseRef(refString)
+
+	if rel != nil && envRef.envName == "" && envRef.version != "" {
+		envRef.orgName = rel.orgName
+		envRef.projectName = rel.projectName
+		envRef.envName = rel.envName
 	}
 
-	return environmentRef{orgName, envName, version}
+	return envRef, nil
 }
 
-func (cmd *envCommand) getEnvRef(args []string) (ref environmentRef, rest []string, err error) {
+// Get an environment reference when creating a new environment
+// If the given path is ambiguous, we need to make an additional API call to disambiguate
+func (cmd *envCommand) getNewEnvRef(
```

Comment:
can we make it more explicit what the difference is between these two functions?